src/main/java/org/prebid/server/vertx/http/CircuitBreakerSecuredHttpClient.java

```diff
@@ -0,0 +1,113 @@
+package org.prebid.server.vertx.http;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.circuitbreaker.CircuitBreaker;
+import io.vertx.circuitbreaker.CircuitBreakerOptions;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.metric.Metrics;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+/**
+ * Wrapper over {@link HttpClient} with circuit breaker functionality.
+ */
+public class CircuitBreakerSecuredHttpClient implements HttpClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(CircuitBreakerSecuredHttpClient.class);
+
+    private final HttpClient httpClient;
+    private final Metrics metrics;
+
+    private final Map<String, CircuitBreaker> circuitBreakerByName = new ConcurrentHashMap<>();
+    private final Function<String, CircuitBreaker> circuitBreakerCreator;
+
+    public CircuitBreakerSecuredHttpClient(Vertx vertx, HttpClient httpClient, Metrics metrics,
+                                           int maxFailures, long timeoutMs, long resetTimeoutMs) {
+        Objects.requireNonNull(vertx);
+        this.httpClient = Objects.requireNonNull(httpClient);
+        this.metrics = Objects.requireNonNull(metrics);
+
+        circuitBreakerCreator = name -> CircuitBreaker.create("http-client-circuit-breaker-" + name,
+                vertx,
+                new CircuitBreakerOptions()
+                        .setMaxFailures(maxFailures)
+                        .setTimeout(timeoutMs)
+                        .setResetTimeout(resetTimeoutMs))
+                .openHandler(ignored -> circuitOpened(name))
+                .halfOpenHandler(ignored -> circuitHalfOpened(name))
+                .closeHandler(ignored -> circuitClosed(name));
+    }
+
+    private void circuitOpened(String name) {
+        logger.warn(String.format("Http client request to %s is failed, circuit opened.", name));
```

Comment:
It is possible to get rid of `String.format()` here and below by resorting to `logger.warn()`s placeholders substitution.